src/Settings/RemoteID.SettingsGroup.json

```diff
@@ -0,0 +1,169 @@
+{
+    "version":              1,
+    "fileType":             "FactMetaData",
+    "QGC.MetaData.Facts":
+[
+{
+    "name":         "enable",
+    "shortDesc":    "Show Drone ID settings page when enabled",
+    "type":         "bool",
+    "default":      false
+},
+{
+    "name":         "operatorID",
+    "shortDesc":    "Operator ID",
+    "longDesc":     "Operator ID. Maximum 20 characters.",
+    "type":         "string",
+    "default":      ""
+},
+{
+    "name":         "operatorIDType",
+    "shortDesc":    "Operator ID type",
+    "type":         "uint8",
+    "enumStrings":  "CAA",
+    "enumValues":   "0",
+    "default":      0
+},
+{
+    "name":         "sendOperatorID",
+    "shortDesc":    "Send Operator ID",
+    "longDesc":     "When enabled, sends operator ID message",
+    "type":         "bool",
+    "default":      false
+},
+{
+    "name":         "selfIDFree",
+    "shortDesc":    "Self ID",
+    "longDesc":     "Optional plain text for operator to specify operations data (Free Text). Maximum 23 characters.",
+    "type":         "string",
+    "default":      ""
+},
+{
+    "name":         "selfIDEmergency",
+    "shortDesc":    "Self ID",
+    "longDesc":     "Optional plain text for operator to specify operations data (Emergency Text). Maximum 23 characters.",
+    "type":         "string",
+    "default":      "Pilot Emergency Status"
+},
+{
+    "name":         "selfIDExtended",
+    "shortDesc":    "Self ID",
+    "longDesc":     "Optional plain text for operator to specify operations data (Extended Text). Maximum 23 characters.",
+    "type":         "string",
+    "default":      ""
+},
+{
+    "name":         "selfIDType",
+    "shortDesc":    "Self ID type",
+    "type":         "uint8",
+    "enumStrings":  "Free Text,Emergency,Extended Status",
+    "enumValues":   "0,1,2",
+    "default":      0
+},
+{
+    "name":         "sendSelfID",
+    "shortDesc":    "Send Self ID",
+    "longDesc":     "When enabled, sends self ID message",
+    "type":         "bool",
+    "default":      false
+},
+{
+    "name":         "basicID",
+    "shortDesc":    "Basic ID",
+    "type":         "string",
+    "default":      ""
+},
+{
+    "name":         "basicIDType",
+    "shortDesc":    "Basic ID Type",
+    "type":         "uint8",
+    "enumStrings":  "None, SerialNumber(ANSI/CTA-2063), CAA, UTM(RFC4122), Specific",
+    "enumValues":   "0,1,2,3,4",
+    "default":      2
+},
+{
+    "name":         "basicIDUaType",
```

Comment:
Isn't this coming from the vehicle?

Reply:
Not totally. This can be set on the parameters of remote id device itself, same as basic id message. If it is set in there, it will work perfectly fine, and in fact that seems to be the preferred way to do so.

But, it can also be set from GCS, and it works as follows:

- If none is set on device, and we set it on GCS, it will appear as basic id 1.
- If one is set on rid device, but we also set it in GCS, it will appear as basic id 2 as long as the type is not serial number.
- If one is set on rid device, but a new one is set in GCS with serial number type, it should override the one set in RID device for basic id 1.